### PR TITLE
Feature/data config UI 670

### PR DESCRIPTION
**Pull request by deepaknevdepsl**
### Description
Data configurations UI

### Issues Resolved
[List any issues this PR will resolve]

### Check List
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: dashboards-observability/public/components/event_analytics/explorer/explorer.tsx [hunk 1]
```diff
@@ -746,6 +746,7 @@ export const Explorer = ({
         handleOverrideTimestamp={handleOverrideTimestamp}
         callback={callbackForConfig}
         changeIsValidConfigOptionState={changeIsValidConfigOptionState}
+        tabID={tabId}
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/explorer.tsx]: any need of passing tabid as prop here as getting it from useContext?

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect, useState } from 'react';
+import { EuiTitle, EuiComboBox, EuiSpacer, EuiButton, EuiFieldText, EuiFlexItem, EuiFormRow, EuiIcon, EuiPanel, EuiText } from '@elastic/eui';
+import { useDispatch, useSelector } from 'react-redux';
+import { render as renderExplorerVis } from '../../../../../../event_analytics/redux/slices/visualization_slice';
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: line 9, 10 can be combined

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+import { selectExplorerVisualization } from '../../../../../../event_analytics/redux/slices/visualization_slice';
+import { AGGREGATION_OPTIONS } from '../../../../../../../../common/constants/explorer';
+import { ButtonGroupItem } from './config_button_group';
+import { visChartTypes } from '../../../../../../../../common/constants/shared';
+
+export const DataConfigPanelItem = ({
+  fieldOptionList,
+  visualizations,
+  tabID
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: same as above .useContext can be used.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'label');
+   };
+
+  const onAggregationChange = (e, index: number, name: string) => {
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'aggregation');
+  };
+
+  const onCustomLabelChange = (e, index: number, name: string) => {
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: can we combine all onfieldOptionChange , onAggregationChange, onCustomLabelChange in to one method as more of it is calling same method? Also , check if we really need this method as 
` onChange={(e) => updateList(label, index, name, sectionName);}` should work.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+  }
+
+  const handleServiceRemove = (index: number, name: string) => {
+    const list = { ...configList };
+    list[name].splice(index, 1);
+    setConfigList(list);
+  };
+
+  const handleServiceAdd = (name: string) => {
+    let newList = { ...configList, [name]: [...configList[name], newEntry] }
```
Comment (abasatwar) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:70]: check 'newEntry' is referring to which variable.

**Review by abasatwar — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+  visualizations,
+}: any) => {
+  const dispatch = useDispatch();
+  const { tabId } = useContext<any>(TabContext);
+  const explorerVisualizations = useSelector(selectExplorerVisualization)[tabId];
+
+  const { data } = visualizations;
+  const { data: vizData = {}, metadata: { fields = [] } = {} } = data?.rawVizData;
+
+  const newEntry = { label: "", aggregation: "", custom_label: "", name: "", side: "right" };
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:25]: Please run eslint. These should be single quotes I think.

File: dashboards-observability/public/components/event_analytics/explorer/explorer.tsx (hunk 1 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/explorer.tsx]: removed from props.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 2 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: Combined import both line

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 7]
```diff
@@ -0,0 +1,212 @@
...
+
+  const [configList, setConfigList] = useState({
+    dimensions: [{ ...newEntry }],
+    metrics: [{ ...newEntry }],
+  });
+
+  useEffect(() => {
+    if (data.rawVizData?.dataConfig) {
+      setConfigList({
+        ...data.rawVizData?.dataConfig
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: The question mark after `rawVizData` can be removed, as you have checked the same in the if condition. The if condition was successful, that is why you are inside this block. i.e. `dataConfig` is defined.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 3 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: removed from props as we are using it from useContext

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 5 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:70]: newEntry variable to add empty state values when user clicks on add button on ui

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 8]
```diff
@@ -0,0 +1,212 @@
...
+    updateList(label, index, name, 'label');
+  };
+
+  const onAggregationChange = (e, index: number, name: string) => {
+    let label = e.length > 0 ? e[0].label : '';
+    updateList(label, index, name, 'aggregation');
+  };
+
+  const onCustomLabelChange = (e, index: number, name: string) => {
+    console.log('name ', name)
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: remove the console message

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 9]
```diff
@@ -0,0 +1,212 @@
...
+    <>
+      <div key={index} className="services">
+        <div className="first-division">
+          <EuiPanel color="subdued">
+            <EuiFormRow
+              label="Aggregation"
+              labelAppend={
+                lists.length !== 1 && (
+                  <EuiText size="xs">
+                    <EuiIcon type="cross" color="danger" onClick={() => handleServiceRemove(index, sectionName)} />
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: This should be multi line. You have 3 properties and you are also including a function inside one of them. Too much easier to read when done vertically.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 10]
```diff
@@ -0,0 +1,212 @@
...
+              label="Custom label"
+            >
+              <EuiFieldText
+                placeholder="Custom label"
+                value={singleField.custom_label}
+                onChange={(e) => updateList(e.target.value, index, sectionName, 'custom_label')}
+                aria-label="Use aria labels when no actual label is in use" />
+            </EuiFormRow>
+
+            {sectionName === 'metrics' && visualizations.vis.name === visChartTypes.Line || visualizations.vis.name === visChartTypes.Bar && (
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: The condition can be saved in a separate variable.- too much logic inside the component JSX.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 11]
```diff
@@ -0,0 +1,212 @@
...
+              <EuiFormRow label="Side">
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:145]: single quotes

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx [hunk 12]
```diff
@@ -0,0 +1,212 @@
...
+                  groupOptions={[{ id: 'left', label: 'Left' }, { id: 'right', label: 'Right' }]}
+                  idSelected={singleField.side || 'right'}
+                  handleButtonChange={(id: string) => updateList(id, index, sectionName, 'side')}
+                />
+              </EuiFormRow>
+            )}
+
+            <EuiSpacer size="s" />
+            {lists.length - 1 === index &&
+              <EuiFlexItem grow={true}>
```
Comment (spattnaik) [inline dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: `={true}` can be removed.

**Review by spattnaik — CHANGES_REQUESTED** [review]

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 6 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:25]: ESLINT is not enforcing.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 7 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: removed question mark from above inner condition

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 8 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: remove handler itself, as this was unused handler

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 9 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: converted this into multiline.

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 10 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: created variable function to check this condition

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 12 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx]: remoted ={true} from this prop

File: dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx (hunk 11 above)
Comment (deepaknevdepsl) [reply dashboards-observability/public/components/event_analytics/explorer/visualizations/config_panel/config_panes/config_controls/data_config_panel_item.tsx:145]: Not changing this as we are using same for other components.

**Review by spattnaik — APPROVED** [review]
Looks good to me. Once Amruta's changes are incorporated, you can move for PR to OpenSearch repo

**Review by abasatwar — APPROVED** [review]

**Comment by deepaknevdepsl** [comment]
As this PR is already merged in sprint1, closing this PR.

**Pull request closed without merging**